lib/mastermind_new.rb

```diff
@@ -0,0 +1,165 @@
+            until user_in.is_valid do
+                user_in = Input.new
+            end
+            current_code = Code.new(user_in.user_input)
+            previous_guesses << current_code.convert
+            pins_out = game_code.compare(current_code.value)
+            remaining_guesses -= 1
+        end
+        puts "The code was: " + game_code.convert.join(' ')
+    end
+
+    class GeneratedList
+        attr_reader :get
+    
+        def initialize
+            @get = []
+            starting_point = { fourth: 1, third: 1, second: 1, first: 1 }
+            until starting_point == { fourth: 6, third: 6, second: 6, first: 6 }
+                @get << starting_point.values
+                starting_point[:first] += 1
+                starting_point.keys.reverse.map.with_index do |pos, i|
+                    if starting_point[pos] > 6
+                        starting_point[pos] = 1
+                        nextval = starting_point.keys.reverse[i + 1]
+                        starting_point[nextval] += 1
+                    end
+                end
+            end
+            @get << { fourth: 6, third: 6, second: 6, first: 6 }.values
+        end
+    
+        def random_code
+            @get[rand(1296)]
+        end
+    end
+    
+    class Input
+        attr_reader :user_input, :is_valid, :output_code
+    
+        def initialize
+            puts 'Please enter a code or command...'
+            print ">>> "
+            validate(gets.chomp)
+        end
+    
+        def validate(check_me)
+            if /[\/]\w/ === check_me
+                commands(check_me.tr('/', '').upcase)
+            elsif /\A[rgybmcRGYBMC][rgybmcRGYBMC][rgybmcRGYBMC][rgybmcRGYBMC]\z/ === check_me
+                @user_input = check_me.split(//)
+                @is_valid = true
+            elsif /\A[rgybmcRGYBMC]\s[rgybmcRGYBMC]\s[rgybmcRGYBMC]\s[rgybmcRGYBMC]\z/ === check_me
+                @user_input = check_me.split(' ')
+                @is_valid = true
+            else
+                puts "Bad Input, make sure the code you enter follows the guidelines."
+                @is_valid = false
+            end
+        end
+    
+        def commands(command)
+            if command == 'HELP'
+                puts 'Help placeholder...'
+            elsif command == 'RESTART'
+                puts 'Restart placeholder...'
+            elsif command == 'QUIT'
+                puts 'Quit placeholder...'
+            else
+                puts 'Bad Command'
+            end
+        end
+    end
+    
+    class Code
+        attr_reader :value, :matched
+    
+        def initialize(new_code)
+            if /[rgybmcRGYBMC]/ === new_code[0]
+                new_code = new_code.join.upcase.split(//)
+                @value = convert(new_code)
+            else
+                @value = new_code
+            end
+        end
+    
+        def convert(input = @value)
+            converted_code = []
+            conversion_container = {
```

Comment:
Probably could be turned into a variable constant at the top of the file like

CONVERSION_CONTAINER = {....